src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -100,16 +100,21 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
 		if (query != null) {
-			boolQB.should(QueryBuilders.matchQuery("name", query).boost(4.0f));
-			boolQB.should(QueryBuilders.prefixQuery("_all", query).boost(1.5f));
-			boolQB.should(QueryBuilders.matchPhrasePrefixQuery("_all", query).slop(2).fuzziness("AUTO"));
+			//Following clause return a module whose UID matches exactly
```

Comment:
I suggest you have an equivalent comment for each one of these lines, describing what it does in plain English.

And, add some newlines, so it ends up like

```
// exact id match scores the highest
boolQB.should(...)

// next comment
boolQB.should(...)
```